hs/src/Reach/Eval/Core.hs

```diff
@@ -4939,11 +4983,23 @@ doApiCall lhs (ApiCallRec {..}) = do
           [jsConst a returnLVal doLog, callOnly [jsThunkStmts a [es interactOut]]]
   let assignLhs = jsConst a lhs $ jsArrayLiteral a [dom, apiReturn]
   let setDetails = jsCall a (jid ".setApiDetails") [slac_who, dom]
-  let ss = [callOnly [onlyThunk], es pub4, assignLhs, es setDetails]
+  let ss = [callOnly [onlyThunk], es pub5, assignLhs, es setDetails]
   return ss
 
-doForkAPI2Case :: [JSExpression] -> App [JSExpression]
-doForkAPI2Case args = do
+prependFunStmts :: [JSStatement] -> JSExpression -> App JSExpression
+prependFunStmts stmts f = do
+  (fargs, fs) <- deconstructFunStmts f
+  return $ jsArrowStmts a fargs $ stmts <> fs
+  where a = jsa f
+
+prependFunArgs :: [JSExpression] -> JSExpression -> App JSExpression
+prependFunArgs xtra f = do
+  (args, stmts) <- deconstructFunStmts f
+  return $ jsArrowStmts a (xtra <> args) stmts
+  where a = jsa f
```

Comment:
I think that these should be the same function or at least the same impl and you just pass `mempty` to one of the options